feature/638

AnimeOneLove2190 opened the pull request:
Add ValidationService

AnimeOneLove2190 committed:
feature/638
Add ValidationService

AnimeOneLove2190 committed:
ValidationService completed;
Add IValidationServiceInterface;
Added dependencies to startup;
The architecture of GhenreService has been changed

AnimeOneLove2190 committed:
Validation service has been modified and changed
The architectures of HallService, MovieService, PlaceServcie, RowService, SessionService and TicketService have been changed

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/HallService.cs:
```diff
@@ -21,38 +21,36 @@ public class HallService : IHallService
+                validationService.ValidationSameNameAlreadyExist(hallFromDB, hallCreate.Name);
             }
             var negativeNumbers = hallCreate.RowsNumbers.Where(x => x <= 0).ToList();
             if (negativeNumbers.Count > 0)
```
use validationService

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/HallService.cs:
```diff
@@ -236,6 +201,7 @@ public async Task DeleteAsync(int id)
                 {
                     var soldTickets = place.Tickets.Where(x => x.IsSold).ToList();
+                    validationService.ValidationEntityHasSoldTickets(nameof(Hall), soldTickets);
                     if (soldTickets.Count > 0)
```
validation exist

artemlipinskiy commented on CinemaTicket.BusinessLogicServices/TicketService.cs:
```diff
@@ -101,38 +82,13 @@ public async Task CreateAsync(TicketCreate ticketCreate)
-            }
+            validationService.ValidationNotFound(sessionFromDB, ticketUpdate.SessionId);
             var rowFromDB = await rowDataAccess.GetRowAsync(placeFromDB.RowId);
             if (sessionFromDB.HallId != rowFromDB.HallId)
```
use validationService

AnimeOneLove2190 committed:
Add new methods in ValidationService
The necessary changes have been made to HallService and TicketService
Added a new field to the ExceptionMessageTemplate
